pkg/registry/core/secret/strategy.go

```diff
@@ -67,12 +71,26 @@ func (strategy) AllowCreateOnUpdate() bool {
 }
 
 func (strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newSecret := obj.(*api.Secret)
+	oldSecret := old.(*api.Secret)
+	dropDisabledFields(newSecret, oldSecret)
 }
 
 func (strategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateSecretUpdate(obj.(*api.Secret), old.(*api.Secret))
 }
 
+func dropDisabledFields(secret *api.Secret, oldSecret *api.Secret) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ImmutableEphemeralVolumes) && oldSecret == nil {
```

Comment:
It seems we only need to call DefaultFeatureGate.Enabled once in this func

Reply:
Technically probably true, but I think this one is easier to reason about in my opinion (for insignificant performance drop).

Reply:
Most times I see this call an "is in use" funtion that encapsulates both old == nil and old.field == nil.